packages/odata-service-inquirer/src/prompts/datasources/sap-system/system-selection/prompt-helpers.ts

```diff
@@ -252,7 +295,18 @@ export async function createSystemChoices(
             };
         }
     } else {
-        const backendSystems = await new SystemService(LoggerHelper.logger).getAll();
+        // Fetch backend systems from systems.json file, credential are fetched once the system is selected
+        const fileSystemStore = getFilesystemStore<BackendSystem>(LoggerHelper.logger);
```

Comment:
Accessing the file store should only be done in the `store` module. I think all the migration code should be there as its not relevant to odata-service-inquirer and what about other users of the `store` like SAP Systems Panel. It would not benefit from the migration at all.

Reply:
It would if the SAP Systems panel also takes the same approach and uses the file for the intial loading of systems